stacks-common/src/types/mod.rs

```diff
@@ -121,75 +122,89 @@ pub struct CoinbaseInterval {
+// | Current            | -              | -                   | 1,552,452,847    | 1000       | -                |
+// | 1st                |   945,000      |   278,950           | 1,627,352,847    | 500 (50%)  | 3.23%            |
+// | 2nd                | 1,050,000      |   383,950           | 1,679,852,847    | 250 (50%)  | 1.57%            |
+// | 3rd                | 1,260,000      |   593,950           | 1,732,352,847    | 125 (50%)  | 0.76%            |
+// | 4th                | 1,470,000      |   803,950           | 1,758,602,847    | 62.5 (50%) | 0.37%            |
+// | -                  | 2,197,560      | 1,531,510           | 1,804,075,347    | 62.5 (0%)  | 0.18%            |
+//
+// The above is for mainnet, which has a burnchain year of 52596 blocks and starts at burnchain height 666050.
 
 /// Mainnet coinbase intervals, as of SIP-029
-pub const COINBASE_INTERVALS_MAINNET: LazyCell<[CoinbaseInterval; 5]> = LazyCell::new(|| {
-    let emissions_schedule = [
-        CoinbaseInterval {
-            coinbase: 1_000 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 0,
-        },
-        CoinbaseInterval {
-            coinbase: 500 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 278_950,
-        },
-        CoinbaseInterval {
-            coinbase: 250 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 383_950,
-        },
-        CoinbaseInterval {
-            coinbase: 125 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 593_950,
-        },
-        CoinbaseInterval {
-            coinbase: (625 * u128::from(MICROSTACKS_PER_STACKS)) / 10,
-            effective_start_height: 803_950,
-        },
-    ];
-    assert!(CoinbaseInterval::check_order(&emissions_schedule));
-    emissions_schedule
-});
-
-/// Testnet coinbase intervals, as of SIP-029
-pub const COINBASE_INTERVALS_TESTNET: LazyCell<[CoinbaseInterval; 5]> = LazyCell::new(|| {
-    let emissions_schedule = [
-        CoinbaseInterval {
-            coinbase: 1_000 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 0,
-        },
-        CoinbaseInterval {
-            coinbase: 500 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 77_777,
-        },
-        CoinbaseInterval {
-            coinbase: 250 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 77_777 * 7,
-        },
-        CoinbaseInterval {
-            coinbase: 125 * u128::from(MICROSTACKS_PER_STACKS),
-            effective_start_height: 77_777 * 14,
-        },
-        CoinbaseInterval {
-            coinbase: (625 * u128::from(MICROSTACKS_PER_STACKS)) / 10,
-            effective_start_height: 77_777 * 21,
-        },
-    ];
-    assert!(CoinbaseInterval::check_order(&emissions_schedule));
-    emissions_schedule
-});
+// This static value is lazily initialized using `OnceLock` to avoid unnecessary
```

Comment:
What was the issue with `LazyCell` here? Using `LazyCell` (or `LazyLock` if you need thread safety) is more concise than `OnceLock`

Reply:
Yes. If we used lazylock it caused errors related to mutability. I added the comment to better explain.

Reply:
I don't mind using lazyLock but once lock is the recommended over lazycell and lazylock. I can be persuaded otherwise but its the recommended standard.

Reply:
You have the same interior mutability issues with `OnceLock` as `LazyCell`, as both provide a `get_mut()` function.

The reason the warning went away is that you changed `COINBASE_INTERVALS_MAINNET` from `const` to `static`, so now it doesn't trigger Clippy's `declare_interior_mutable_const` because there's no `const`

I'd stick with the `LazyCell`, it's cleaner and more efficient (not thread-safe, so no spinlock or mutex required to access it). Just make it `static` to kill the warning

Reply:
You cannot make it static. ````UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>` cannot be shared between threads safely
within `LazyCell<[CoinbaseInterval; 5]>`, the trait `Sync` is not implemented for `UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>`, which is required by `LazyCell<[CoinbaseInterval; 5]>: Sync`
shared static variables must have a type that implements `Sync
```

Reply:
I cannot make it static otherwise I have these issues. This was the only way I could seem to get around them.

````rust
UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>` cannot be shared between threads safely
within `LazyCell<[CoinbaseInterval; 5]>`, the trait `Sync` is not implemented for `UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>`, which is required by `LazyCell<[CoinbaseInterval; 5]>: Sync`
shared static variables must have a type that implements `Sync

Reply:
I cannot make it static. 

```rust
`UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>` cannot be shared between threads safely
within `LazyCell<[CoinbaseInterval; 5]>`, the trait `Sync` is not implemented for `UnsafeCell<cell::lazy::State<[CoinbaseInterval; 5], fn() -> [CoinbaseInterval; 5]>>`, which is required by `LazyCell<[CoinbaseInterval; 5]>: Sync`
shared static variables must have a type that implements `Sync

I could just silence the warning if you prefer? But I cannot just declare it static and still use LazyCell

Reply:
That error means it's shared by multiple threads and must use a `Sync` type, like `LazyLock`. `OnceLock` works also, it's just cumbersome and awkward to work with. We use `LazyLock` elsewhere in the code, it's basically the modern stdlib replacement for `lazy_static!`. See also PR #5055

Reply:
Done.